dojo/settings/settings.dist.py

```diff
@@ -446,375 +475,427 @@ def generate_url(scheme, double_slashes, user, password, host, port, path, param
-
-GITHUB_ENTERPRISE_OAUTH2_ENABLED = env('DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_OAUTH2_ENABLED')
-SOCIAL_AUTH_GITHUB_ENTERPRISE_URL = env('DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_URL')
-SOCIAL_AUTH_GITHUB_ENTERPRISE_API_URL = env('DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_API_URL')
-SOCIAL_AUTH_GITHUB_ENTERPRISE_KEY = env('DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_KEY')
-SOCIAL_AUTH_GITHUB_ENTERPRISE_SECRET = env('DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_SECRET')
-
-DOCUMENTATION_URL = env('DD_DOCUMENTATION_URL')
+    SOCIAL_AUTH_GITLAB_SCOPE += ["read_repository"]
+
+AUTH0_OAUTH2_ENABLED = env("DD_SOCIAL_AUTH_AUTH0_OAUTH2_ENABLED")
+SOCIAL_AUTH_AUTH0_KEY = env("DD_SOCIAL_AUTH_AUTH0_KEY")
+SOCIAL_AUTH_AUTH0_SECRET = env("DD_SOCIAL_AUTH_AUTH0_SECRET")
+SOCIAL_AUTH_AUTH0_DOMAIN = env("DD_SOCIAL_AUTH_AUTH0_DOMAIN")
+SOCIAL_AUTH_AUTH0_SCOPE = env("DD_SOCIAL_AUTH_AUTH0_SCOPE")
+SOCIAL_AUTH_TRAILING_SLASH = env("DD_SOCIAL_AUTH_TRAILING_SLASH")
+
+KEYCLOAK_OAUTH2_ENABLED = env("DD_SOCIAL_AUTH_KEYCLOAK_OAUTH2_ENABLED")
+SOCIAL_AUTH_KEYCLOAK_KEY = env("DD_SOCIAL_AUTH_KEYCLOAK_KEY")
+SOCIAL_AUTH_KEYCLOAK_SECRET = env("DD_SOCIAL_AUTH_KEYCLOAK_SECRET")
+SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY = env("DD_SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY")
+SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL = env(
+    "DD_SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL"
+)
+SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL = env("DD_SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL")
+SOCIAL_AUTH_KEYCLOAK_LOGIN_BUTTON_TEXT = env(
+    "DD_SOCIAL_AUTH_KEYCLOAK_LOGIN_BUTTON_TEXT"
+)
+
+GITHUB_OAUTH2_ENABLED = env("DD_SOCIAL_AUTH_GITHUB_OAUTH2_ENABLED")
+
+# github ouath2 settings
+SOCIAL_AUTH_GITHUB_KEY = env("DD_SOCIAL_AUTH_GITHUB_KEY")
+SOCIAL_AUTH_GITHUB_SECRET = env("DD_SOCIAL_AUTH_GITHUB_SECRET")
+
+# github org oauth2
+SOCIAL_AUTH_GITHUB_ORG_KEY = env("DD_SOCIAL_AUTH_GITHUB_KEY")
+SOCIAL_AUTH_GITHUB_ORG_SECRET = env("DD_SOCIAL_AUTH_GITHUB_SECRET")
+SOCIAL_AUTH_GITHUB_ORG_NAME = env("DD_SOCIAL_AUTH_GITHUB_ORG_NAME")
+
+# github team oauth2
+SOCIAL_AUTH_GITHUB_TEAM_KEY = env("DD_SOCIAL_AUTH_GITHUB_KEY")
+SOCIAL_AUTH_GITHUB_TEAM_SECRET = env("DD_SOCIAL_AUTH_GITHUB_SECRET")
+SOCIAL_AUTH_GITHUB_TEAM_ID = env("DD_SOCIAL_AUTH_GITHUB_TEAM_ID")
+
+if GITHUB_OAUTH2_ENABLED:
+    if SOCIAL_AUTH_GITHUB_TEAM_ID:
+        AUTHENTICATION_BACKENDS += "social_core.backends.github.GithubTeamOAuth2"
+    elif SOCIAL_AUTH_GITHUB_ORG_NAME:
+        AUTHENTICATION_BACKENDS += (
+            "social_core.backends.github.GithubOrganizationOAuth2"
+        )
+    else:
+        AUTHENTICATION_BACKENDS += "social_core.backends.github.GithubOAuth2"
+
+
+GITHUB_ENTERPRISE_OAUTH2_ENABLED = env(
+    "DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_OAUTH2_ENABLED"
+)
+SOCIAL_AUTH_GITHUB_ENTERPRISE_URL = env("DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_URL")
+SOCIAL_AUTH_GITHUB_ENTERPRISE_API_URL = env("DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_API_URL")
+SOCIAL_AUTH_GITHUB_ENTERPRISE_KEY = env("DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_KEY")
+SOCIAL_AUTH_GITHUB_ENTERPRISE_SECRET = env("DD_SOCIAL_AUTH_GITHUB_ENTERPRISE_SECRET")
```

Comment:
Dropping this comment for other reviewers for where the code changes are

Reply:
Apologies, my linters stripped a lot of trailing whitespace. Will look at fixing the merge conflict 🔜